### PR TITLE
fix(extension): fix extension nls config bug

### PR DESCRIPTION
**Pull request by shilin8805**
### Types

- [x] 🐛 Bug Fixes

#2529 这个PR还是有bug，有两行写入languagePackJson的代码被误删除了，setupExtensionNLSConfig这个方法在activate的时候必须放在后面执行，否则不会生效，修改后已经验证没有问题了，帮忙Review后合并到2.24分支吧

image

### <samp>🤖 Generated by Copilot at 767e0d0</samp>

* Comment out the call to `setupExtensionNLSConfig` in the `activate` method of the `ExtensionServiceImpl` class to avoid overwriting the language pack settings by the plugins ([link](https://github.com/opensumi/core/pull/2603/files?diff=unified&w=0#diff-3d705c2bf9f71760b12685e3d9e61fc6339b3886d9ace8c550014b9b2547b806L200-R201))
* Add the logic to write the language pack settings to the `languagepack.json` file in the extension folder using the `fileService` methods in the `ExtensionServiceClientImpl` class ([link](https://github.com/opensumi/core/pull/2603/files?diff=unified&w=0#diff-5637d5b22e87217b5f315ce965f10b7c914d1337fe1ce29ec2503665f446c74aR207-R209))

### <samp>🤖 Generated by Copilot at 767e0d0</samp>

Commented out a method call in `extension.service.ts` and added logic to write language pack settings to `languagepack.json` in `extension.service.client.ts`. These changes fix the issue of incorrect language pack settings when loading extensions.

### CODE REVIEW

File: packages/extension/src/browser/extension.service.ts [hunk 1]
```diff
@@ -197,7 +197,8 @@ export class ExtensionServiceImpl extends WithEventBus implements ExtensionServi
 
   public async activate(): Promise<void> {
     // setup the basic environment
-    await this.setupExtensionNLSConfig();
+    // 不能在这里设置，因为这里的设置会被插件覆盖
+    // await this.setupExtensionNLSConfig();
```
Comment (bytemain) [inline packages/extension/src/browser/extension.service.ts:201]: 如果这里不设置，就不会给插件进程设置 `VSCODE_NLS_CONFIG` 这个变量，某些插件直接被激活，设置的语言就不对。

目前看起来，还是得像 [这个讨论](https://github.com/opensumi/core/discussions/2317#discussioncomment-5052245) 中说的，先激活 language pack

**Review by bytemain — CHANGES_REQUESTED** [review]

File: packages/extension/src/browser/extension.service.ts (hunk 1 above)
Comment (bytemain) [reply packages/extension/src/browser/extension.service.ts:201]: 我先试一下吧，可能是我理解错了

**Comment by erha19** [comment]
@shilin8805 这里 publish 没用，你基于 opensumi/core checkout 一个分支出来提交吧

**Pull request closed without merging**